src/HotChocolate/Language/src/Language.Rewriters/Contracts/ISyntaxNavigator.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+
+namespace HotChocolate.Language.Rewriters;
+
+public interface ISyntaxNavigator
```

Comment:
/// <summary>
/// The syntax navigator keeps track of the syntax path that has been traversed and allows to access the nodes in the path in a streamlined way.
/// </summary>